lib/experimental/new_embed.dart

```diff
@@ -193,6 +206,69 @@ class TestTabView extends TabView {
   }
 }
 
+/// A line of text next to the [ExecuteButton] that reports test result messages
+/// in red or green.
+class TestResultLabel {
+  TestResultLabel(this.element);
+
+  final DivElement element;
+
+  void setResult(TestResult result) {
+    element.text = result.message;
+
+    if (result.success) {
+      element.classes.add('text-green');
+      element.classes.remove('text-red');
+    } else {
+      element.classes.remove('text-green');
+      element.classes.add('text-red');
+    }
+  }
+}
+
+class ExecuteCodeButton {
+  /// This constructor will throw if the provided element has no child with a
+  /// CSS class that begins with "octicon-".
+  ExecuteCodeButton(this.element, VoidCallback onClick) {
+    final iconElement = element.children.firstWhere(Octicon.elementIsOcticon);
+    _icon = Octicon(iconElement);
+    element.addEventListener('click', (e) => onClick());
+  }
+
+  static const readyIconName = 'chevron-right';
```

Comment:
Perhaps instead one of:
- https://octicons.github.com/icon/triangle-right/
- https://octicons.github.com/icon/check/
- https://octicons.github.com/icon/play/

Reply:
A play icon would be best, but AFAICT it doesn't appear in the version of Octicons used in the package @jacob314 published. Updated to triangle-right for now, and I'll ask about upping the version to something newer.